### PR TITLE
Add option to skip updating or recreating comments when message is unchanged

### PR DESCRIPTION
**Pull request by aburgel**
To avoid repeatedly posting the same message, I've adding an option to compare the new and previous message and skip hide_and_recreate, recreate and update steps if they are the same.

### CODE REVIEW

File: src/comment.ts [hunk 1]
```diff
@@ -162,7 +166,7 @@ export async function minimizeComment(
 }
 
 export function getBodyOf(
-  previous: {body?: string},
+  previous: {body: string},
```
Comment (aburgel) [inline src/comment.ts:169]: The body property of IssueComment is always present.

File: src/comment.ts [hunk 2]
```diff
@@ -176,3 +180,12 @@ export function getBodyOf(
...
   return previous.body?.replace(/(<details.*?)\s*\bopen\b(.*>)/g, "$1$2")
 }
+
+export function commentsEqual(
+  body: string,
+  previous: string,
+  header: string
+): boolean {
+  const newBody = bodyWithHeader(body, header)
+  return newBody === previous
```
Comment (aburgel) [inline src/comment.ts:190]: I did some testing, it looks like github doesn't trim whitespace from the start or end of a comment, so I think an equality check like this should work.

**Comment by aburgel** [comment]
This fixes #1023

**Comment by marocchino** [comment]
Thanks!

**Comment by marocchino** [comment]
It tagged as [v2.8.0](https://github.com/marocchino/sticky-pull-request-comment/releases/tag/v2.8.0)

**Pull request merged**